scripts/ci/sync_extensions.py

```diff
@@ -0,0 +1,141 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=broad-except
+
+import os
+import re
+import json
+from subprocess import check_output
+
+DEFAULT_TARGET_INDEX_URL = os.getenv('AZURE_EXTENSION_TARGET_INDEX_URL')
+STORAGE_ACCOUNT_KEY = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT_KEY')
+STORAGE_ACCOUNT = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT')
+STORAGE_CONTAINER = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_CONTAINER')
+
+
+def _get_updated_extension_names():
+    cmd = 'git --no-pager diff --diff-filter=ACMRT HEAD~1 -- src/index.json'
+    updated_content = check_output(cmd.split()).decode('utf-8')
+    updated_urls = [line.replace('+', '') for line in updated_content.splitlines() if line.startswith('+') and not line.startswith('+++') and 'downloadUrl' in line]
+    updated_exts = []
+    NAME_REGEX = r'.*/([^/]*)-\d+.\d+.\d+'
+    if not updated_urls:
+        return updated_exts
+    for line in updated_urls:
+        search_result = re.search(r'"downloadUrl":\s+"(.*?)"', line)
+        if search_result is not None:
+            url = search_result.group(1)
+            extension_name = re.findall(NAME_REGEX, url)[0].replace('_', '-')
+            updated_exts.append(extension_name)
+    return updated_exts
+
+
+def _download_file(url):
+    import requests
+    count = 3
+    the_ex = None
+    while count > 0:
+        try:
+            response = requests.get(url, allow_redirects=True)
```

Comment:
since whl package size may be very big, is this lib supporting resume from break-point? if there any more reliable/dedicated package downloading library?

Reply:
Changed to stream downloading so we can handle big files well.